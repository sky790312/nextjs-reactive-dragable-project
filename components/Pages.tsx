import React from 'react'
import styled from 'styled-components'
import Page from '@/components/Page'
import { useSelector, shallowEqual } from 'react-redux'
import { RootState } from '@/store/index'

const Pages: React.FC = () => {
  const pages = useSelector(
    (state: RootState) => state.page.pages,
    shallowEqual,
  )

  return (
    <PagesWrapper>
      <h4>Pages</h4>
      {pages.map(page => (
        <Page key={page.id} page={page} />
      ))}
    </PagesWrapper>
  )
}

export default Pages

const PagesWrapper = styled.div`
  border-bottom: 1px solid;
  padding-bottom: 16px;
`
