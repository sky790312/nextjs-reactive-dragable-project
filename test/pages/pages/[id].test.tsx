import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import Pages from '@/pages/pages/[id]'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: '',
      query: '',
      asPath: '',
    }
  },
}))

describe('pages/[id] page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Pages />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
