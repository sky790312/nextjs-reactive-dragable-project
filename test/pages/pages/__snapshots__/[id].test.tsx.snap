// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/[id] page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="id__PagesWrapper-sc-1tgfkr4-0 eKtCWS"
  >
    <div
      class="LeftPanel__LeftPanelWrapper-q1m529-0 klFdKt"
    >
      <div
        class="Pages__PagesWrapper-hol3qz-0 fQzYNz"
      >
        <h4>
          Pages
        </h4>
      </div>
      <div
        class="Elements__ElementsWrapper-vrh229-0"
      >
        <h4>
          Elements
        </h4>
      </div>
    </div>
    <div
      class="Canvas__CanvasWrapper-sc-5gmsqo-0 hInTKC"
    />
    <div
      class="RightPanel__RightPanelWrapper-sc-1amlud5-0 dVOEbY"
    >
      <label
        class="RightPanel__Label-sc-1amlud5-1 eIbqoz"
      >
        X 
        <input
          max="999"
          min="0"
          type="number"
          value="0"
        />
      </label>
      <label
        class="RightPanel__Label-sc-1amlud5-1 eIbqoz"
      >
        Y 
        <input
          max="999"
          min="0"
          type="number"
          value="0"
        />
      </label>
      <label
        class="RightPanel__Label-sc-1amlud5-1 eIbqoz"
      >
        O 
        <input
          max="100"
          min="0"
          type="number"
          value="1"
        />
        <input
          max="100"
          min="0"
          type="range"
          value="100"
        />
      </label>
      <label
        class="RightPanel__Label-sc-1amlud5-1 eIbqoz"
      >
        B 
        <div
          class="ColorPicker__ColorPickerWrapper-sc-1fchp4r-0 eSLzde"
          color="#008800"
        >
          <input
            class="ColorPicker__ColorInput-sc-1fchp4r-1 gufyRf"
            type="color"
            value="#008800"
          />
        </div>
        #008800
      </label>
    </div>
  </div>
</DocumentFragment>
`;
